lib/SystemPruneCaches.js

```diff
@@ -0,0 +1,161 @@
+const { readdir: _readdir, stat: _stat } = require('fs');
+const { basename, join } = require('path');
+
+const _rimraf = require('rimraf');
+
+const logMessages = require('./util/log-messages');
+const pluginCompat = require('./util/plugin-compat');
+const promisify = require('./util/promisify');
+
+const readdir = promisify(_readdir);
+const rimraf = promisify(_rimraf);
+const stat = promisify(_stat);
+
+const directorySize = async dir => {
+  const _stat = await stat(dir);
+  if (_stat.isFile()) {
+    return _stat.size;
+  }
+
+  if (_stat.isDirectory()) {
+    const names = await readdir(dir);
+    let size = 0;
+    for (const name of names) {
+      size += await directorySize(join(dir, name));
+    }
+    return size;
+  }
+
+  return 0;
+};
+
+class CacheInfo {
+  constructor(id = '') {
+    this.id = id;
+    this.lastModified = 0;
+    this.size = 0;
+  }
+
+  static async fromDirectory(dir) {
+    const info = new CacheInfo(basename(dir));
+    info.lastModified = +(await stat(join(dir, 'stamp'))).mtime;
```

Comment:
Why do you need the `+` here? Isn't this returning a date string?

Reply:
It's a Date object. And the `+` is implicitly casting it into a Number. I should do better and use `Number(...)` or `.getTime()`